### ISSUE TITLE
Vision Transformer

### ISSUE THREAD

**Issue by seanshpark** [seanshpark/study#1]
About [Visiton Transformer](https://github.com/google-research/vision_transformer) from google research.

**Comment by seanshpark** [comment]
Papers
- Vision Transformer: https://arxiv.org/abs/2010.11929
- MLP-Mixer: https://arxiv.org/abs/2105.01601
- How to train your ViT: https://arxiv.org/abs/2106.10270
- When Vision Transformers Outperform ResNets without Pretraining or Strong Data Augmentations: https://arxiv.org/abs/2106.01548

**Comment by seanshpark** [comment]
## Try ViT on colab

<a href="https://colab.research.google.com/github/google-research/vision_transformer/blob/main/vit_jax.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

(copy from vit_jax.ipynb)

**Comment by seanshpark** [comment]
## Convert google ViT to tflite ?

what is `jax` ?

**Comment by seanshpark** [comment]
## ViT packages

https://github.com/lucidrains/vit-pytorch

**Comment by seanshpark** [comment]
https://github.com/s-chh/PyTorch-Scratch-Vision-Transformer-ViT

default target is `mnist`, the numbers
- goes to 99%.

to check with FashionMNIST, run `python3 main.py --dataset=fashionmnist`
- goes to 92%

```
Train acc: 93.18%	Train loss: 0.1845
Train Confusion Matrix:
[[5319    3   60  113    8    1  474    0   12    0]
 [   1 5930    1   46    7    0    5    0    1    0]
 [  64    2 5362   39  254    0  268    0    3    0]
 [  75   27   17 5668  128    0   72    0    3    1]
 [   6    5  221  108 5272    0  367    0    8    1]
 [   0    0    0    0    0 5806    0  128    2   50]
 [ 474    5  258  115  251    0 4880    0    6    0]
 [   0    0    0    0    0   26    0 5855    1  114]
 [   9    0    6    4    7    9   10    4 5939    3]
 [   0    0    0    1    1   21    0  177    0 5790]]
Test acc: 92.19%	Test loss: 0.2199
Test Confusion Matrix:
[[862   2  15  22   2   0  93   0   4   0]
 [  1 990   0   7   1   0   0   0   1   0]
 [ 17   0 889  12  27   0  54   0   1   0]
 [ 16   6   6 919  28   0  24   0   1   0]
 [  1   0  34  19 883   0  62   0   1   0]
 [  0   0   0   0   0 979   0  14   0   7]
 [ 90   2  43  16  68   0 776   0   5   0]
 [  0   0   0   0   0   9   0 977   0  14]
 [  5   1   2   2   2   0   1   1 986   0]
 [  1   0   0   0   0   3   0  38   0 958]]
Best test acc: 92.28%
```

**Comment by seanshpark** [comment]
With https://medium.com/correll-lab/building-a-vision-transformer-model-from-scratch-a3054f707cc6
- sometimes max test score is about 70%
- sometimes test score goes back to 10%

difference with schh_vit (92%) ?
- add more LayerNorm
- add more Dropout
- use AdamW

**Comment by seanshpark** [comment]
AdamW ?
- https://arxiv.org/pdf/1711.05101
- DECOUPLED WEIGHT DECAY REGULARIZATION
